scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,898 @@
+------------------------------------
+-- Egg Hunt Egg-Stravaganza
+------------------------------------
+require("scripts/globals/settings")
+require("scripts/globals/zone")
+require("scripts/globals/utils")
+require("scripts/globals/npc_util")
+------------------------------------
+xi = xi or {}
+xi.events = xi.events or {}
+xi.events.egg_hunt = xi.events.egg_hunt or {}
+xi.events.egg_hunt.data = xi.events.egg_hunt.data or {}
+xi.events.egg_hunt.entities = xi.events.egg_hunt.entities or {}
+
+local event = SeasonalEvent:new("egg_hunt")
+
+-- Default settings
+local settings =
+{
+    START  = { DAY   =  6, MONTH = 4 },
+    FINISH = { DAY   = 17, MONTH = 4 },
+
+    VAR =
+    {
+        DAILY_EGG     = "[EGG_HUNT]DAILY_EGG",
+        DAILY_REWARD  = "[EGG_HUNT]DAILY_REWARD",
+        DAILY_BONUS   = "[EGG_HUNT]DAILY_BONUS",
+        DAILY_HELM    = "[EGG_HUNT]DAILY_HELM",
+        FIRST_THREE   = "[EGG_HUNT]FIRST_THREE",
+    },
+
+    -- Default era is 2005
```

Comment:
Not blocking this PR, but I wonder if there's a way we could consolidate this in the future with other events

Reply:
Thanks for your suggestions @claywar! I'm happy to follow up on these recommendations if you guys don't mind waiting on the merge.